inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -0,0 +1,351 @@
+    const size_t lastIdx = nDims - 1;
+
+    order.resize(nDims);
+    optimizedParams.src_block_dims.resize(nDims);
+    order[0] = 0;
+    optimizedParams.src_block_dims[0] = srcDims[0];
+
+    auto orderAndReshape = [&](const size_t idx1, const size_t idx2, const size_t shift, const SizeVector& dims) {
+        for (size_t i = 0; i < nSpatialDims; i++) {
+            order[i * 2 + shift] = i + idx1;
+            order[i * 2 + shift + 1] = i + idx2;
+
+            optimizedParams.src_block_dims[order[i * 2 + shift]] = dims[i + shift];
+            optimizedParams.src_block_dims[order[i * 2 + shift + 1]] = blockSize;
+        }
+    };
+
+    if (isBlocked) {
+        SizeVector srcBlockedDims = getParentEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+        SizeVector dstBlockedDims = getChildEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+
+        size_t orderShiftForBlocks, orderShiftForDims;
+        if (mode == Mode::BLOCKS_FIRST) {
+            orderShiftForBlocks = 1;
+            orderShiftForDims = nSpatialDims + 2;
+
+            optimizedParams.src_block_dims[nSpatialDims + 1] = srcBlockedDims[1] / blockStep;
+            optimizedParams.src_block_dims[lastIdx] = srcBlockedDims.back();
+
+            order[1] = nSpatialDims + 1;
+            order[lastIdx] = lastIdx;
+        } else {
+            orderShiftForBlocks = nSpatialDims + 4;
+            orderShiftForDims = 3;
+
+            size_t newBlockSize = srcBlockedDims.back() / blockStep;
+            size_t newBlocksCount = srcBlockedDims[1] * newBlockSize / srcBlockedDims.back();
+            optimizedParams.src_block_dims[1] = newBlocksCount;
+            optimizedParams.src_block_dims[2] = srcBlockedDims[1] / newBlocksCount;
+            optimizedParams.src_block_dims[lastIdx - nSpatialDims] = newBlockSize;
+
+            order[1] = 1;
+            order[2] = 3;
+            order[lastIdx - 1] = 2;
+            order[lastIdx] = lastIdx - nSpatialDims;
+        }
+
+        orderAndReshape(orderShiftForDims, orderShiftForBlocks, 2, srcBlockedDims);
+    } else if (getParentEdgeAt(0)->getMemory().GetDesc().isTailCFormat()) {
+        srcDims.push_back(srcDims[1]);
+        dstDims.push_back(dstDims[1]);
+        srcDims.erase(srcDims.begin() + 1);
+        dstDims.erase(dstDims.begin() + 1);
+
+        size_t shift = static_cast<size_t>(mode == DEPTH_FIRST) + nSpatialDims + 1;
+        order[lastIdx] = mode == Mode::DEPTH_FIRST ? nSpatialDims + 1 : lastIdx;
+        optimizedParams.src_block_dims[order[lastIdx]] = srcDims.back() / blockStep;
+
+        orderAndReshape(1, shift, 1, srcDims);
+    } else {
+        size_t shift = static_cast<size_t>(mode == DEPTH_FIRST) + 1;
+        order[1] = mode == DEPTH_FIRST ? 1 : nSpatialDims + 1;
+        optimizedParams.src_block_dims[order[1]] = srcDims[1] / blockStep;
+
+        orderAndReshape(nSpatialDims + 2, shift, 2, srcDims);
```

Comment:
try to give a name for hard coded 2 such as firstSpatialOrder or smt?

Reply:
Ok, I added this variable. Thank you!